### PR TITLE
fix and refactor KHI growth test

### PR DESCRIPTION
**Pull request by psychocoderHPC**
- fix python script error if the error log file does not exist (create the file if it does not exist)
- change tool parameters of validate.sh
- change tool parameters of ci.sh
- give ci.sh and validate.sh executable rights

### CODE REVIEW

**Comment by PrometheusPi** [comment]
@psychocoderHPC please rebase and push again.

File: lib/python/test/KHI_growthRate/DifferenceKHIGrowthRate.py [hunk 1]
```diff
@@ -90,4 +90,4 @@ def getDifferenceInPercentage(theoryMax: float, simMax):
         """
 
         simMax = max(simMax)
-        return (theoryMax - simMax) / simMax * 100
+        return (theoryMax - simMax) / theoryMax * 100
```
Comment (PrometheusPi) [inline lib/python/test/KHI_growthRate/DifferenceKHIGrowthRate.py:93]: as discussed offline this should be correct - @MSVoss any objections?

File: share/picongpu/tests/KHI_growthRate/README.rst [hunk 2]
```diff
@@ -0,0 +1,31 @@
+KelvinHelmholtz: Kelvin-Helmholtz Instability
```
Comment (PrometheusPi) [inline share/picongpu/tests/KHI_growthRate/README.rst]: Not really needed (file duplication)

File: share/picongpu/tests/KHI_growthRate/include/picongpu/param/particleFilters.param [hunk 3]
```diff
@@ -0,0 +1,119 @@
+/* Copyright 2013-2022 Rene Widera
```
Comment (PrometheusPi) [inline share/picongpu/tests/KHI_growthRate/include/picongpu/param/particleFilters.param:1]: not needed - can be removed - why do we actually have this in the KHI example?

File: share/picongpu/tests/KHI_growthRate/include/picongpu/param/png.param [hunk 4]
```diff
@@ -0,0 +1,100 @@
+/* Copyright 2013-2022 Axel Huebl, Heiko Burau, Rene Widera, Richard Pausch
```
Comment (PrometheusPi) [inline share/picongpu/tests/KHI_growthRate/include/picongpu/param/png.param:1]: not needed for the CI - there is no png output

**Review by PrometheusPi — CHANGES_REQUESTED** [review]

File: share/picongpu/tests/KHI_growthRate/include/picongpu/param/radiationObserver.param [hunk 5]
```diff
@@ -0,0 +1,86 @@
+/* Copyright 2013-2022 Heiko Burau, Rene Widera, Richard Pausch
```
Comment (PrometheusPi) [inline share/picongpu/tests/KHI_growthRate/include/picongpu/param/radiationObserver.param]: not needed for CI

File: share/picongpu/tests/KHI_growthRate/include/picongpu/param/speciesDefinition.param [hunk 6]
```diff
@@ -0,0 +1,106 @@
+/* Copyright 2013-2022 Rene Widera, Benjamin Worpitz, Heiko Burau
```
Comment (PrometheusPi) [inline share/picongpu/tests/KHI_growthRate/include/picongpu/param/speciesDefinition.param]: not needed for CI - we can use default

**Review by PrometheusPi — CHANGES_REQUESTED** [review]

**Comment by BrianMarre** [comment]
Short question why are the KHI growth test located in `lib/python/test/` and not in `test` where they belong.

Does not necessarily need to be fixed in this pull request, but should be corrected eventually.

File: share/picongpu/tests/KHI_growthRate/include/picongpu/param/particleFilters.param (hunk 3 above)
Comment (psychocoderHPC) [reply share/picongpu/tests/KHI_growthRate/include/picongpu/param/particleFilters.param:1]: this file contains the selection of the particle moving up and down.

File: share/picongpu/tests/KHI_growthRate/include/picongpu/param/png.param (hunk 4 above)
Comment (psychocoderHPC) [reply share/picongpu/tests/KHI_growthRate/include/picongpu/param/png.param:1]: I would keep it in if we need to debug the code

**Comment by psychocoderHPC** [comment]
> Short question why are the KHI growth test located in `lib/python/test/` and not in `test` where they belong.
> 
> Does not necessarily need to be fixed in this pull request, but should be corrected eventually.

I do not know. In general this code could be in the test case only because it is test case specific. This is something we can rework in a separate PR.

**Comment by psychocoderHPC** [comment]
@PrometheusPi I updated the PR

**Comment by MSVoss** [comment]
If the code should be in a different folder, we can also integrate it directly into the pull request Unified Testsuite. Since a revision is taking place there anyway.

**Comment by BrianMarre** [comment]
> If the code should be in a different folder, we can also integrate it directly into the pull request Unified Testsuite. Since a revision is taking place there anyway.

Ideally it should be next to pypicongpu and picmi tests in `test/...`

**Comment by psychocoderHPC** [comment]
> If the code should be in a different folder, we can also integrate it directly into the pull request Unified Testsuite. Since a revision is taking place there anyway.

Let us keep the code where it is until my and our PR are in and then let us move the code.
If we move and change the code in one PR the diff will unreadable.

**Review by PrometheusPi — APPROVED** [review]

**Pull request merged**